From 8e38ecc593ab0b1c68ab5cfcef02063f46021f2b Mon Sep 17 00:00:00 2001
From: sudodoki <smd.deluzion@gmail.com>
Date: Thu, 30 Jan 2014 14:57:34 +0200
Subject: [PATCH] Add license header for coffee & js files

---
 Gruntfile.js | 32 +++++++++++++++++++++++---------
 package.json |  1 +
 2 files changed, 24 insertions(+), 9 deletions(-)

diff --git a/Gruntfile.js b/Gruntfile.js
index 046b67d..20eed2b 100644
--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -2,6 +2,7 @@ module.exports = function (grunt) {
   require('matchdep').filterDev('grunt-*').forEach(grunt.loadNpmTasks);
 
   grunt.initConfig({
+    pkg: grunt.file.readJSON('package.json'),
     coffee: {
       lib: {
         options: { bare: false },
@@ -17,15 +18,25 @@ module.exports = function (grunt) {
       },
     },
     concat: {
-      'build/morris.coffee': [
-        'lib/morris.coffee',
-        'lib/morris.grid.coffee',
-        'lib/morris.hover.coffee',
-        'lib/morris.line.coffee',
-        'lib/morris.area.coffee',
-        'lib/morris.bar.coffee',
-        'lib/morris.donut.coffee'
-      ],
+      'build/morris.coffee': {
+        options: {
+          banner: "### @license\n"+
+                  "<%= pkg.name %> v<%= pkg.version %>\n"+
+                  "Copyright <%= (new Date()).getFullYear() %> <%= pkg.author.name %> All rights reserved.\n" +
+                  "Licensed under the <%= pkg.license %> License.\n" +
+                  "###\n",
+        },
+        src: [
+          'lib/morris.coffee',
+          'lib/morris.grid.coffee',
+          'lib/morris.hover.coffee',
+          'lib/morris.line.coffee',
+          'lib/morris.area.coffee',
+          'lib/morris.bar.coffee',
+          'lib/morris.donut.coffee'
+        ],
+        dest: 'build/morris.coffee'
+      },
       'build/spec.coffee': ['spec/support/**/*.coffee', 'spec/lib/**/*.coffee']
     },
     less: {
@@ -39,6 +50,9 @@ module.exports = function (grunt) {
     },
     uglify: {
       build: {
+        options: {
+          preserveComments: 'some'
+        },
         files: {
           'morris.min.js': 'morris.js'
         }
diff --git a/package.json b/package.json
index ed481ae..5ecaa11 100644
--- a/package.json
+++ b/package.json
@@ -2,6 +2,7 @@
   "name": "morris.js",
   "version": "0.5.0",
   "homepage": "http://oesmith.github.com/morris.js",
+  "license": "BSD-2-Clause",
   "description": "Easy, pretty charts",
   "author": {
     "name": "Olly Smith",
